src/sim/controller/ControllerClass.py

```diff
@@ -21,24 +32,36 @@ def addNewRobot(self, x, y):
         # Add robot to model
         self.__model.addRobot(robotModel)
         
-        # Start Threads for Robot (v_main for robot with ip)
-        
         # Update the View
         robotItem = self.__view.drawRobot(robotModel, x, y)   
 
         # Update the robotItem Field
         robotModel.robotItem = robotItem
 
+        # Add robot positions to known list of robot positions
+        self.__vg.robot_positions.update({robotModel.ip: (x, y)})
+
+        # Start Threads for Robot (v_main for robot with ip)
+        v_main_thread = KillableThread(v_main, (robotModel, self.__vg), name=robotModel.ip)
+        self.__threadList.append(v_main_thread)
+        v_main_thread.start()
+
+
     def deleteRobots(self, robotItems):
         for robotItem in robotItems:
             robotModel = next((x for x in self.__model.robots if x.robotItem is robotItem), None)
 
             if robotModel is None:
                 raise "Error in deleteRobots(). Robot is not in list"
             
-            # Stop the Threads
-            robotModel.thread.kill()
-            robotModel.thread.join()
+            for thread in self.__threadList:
```

Comment:
why would we disassociate the thread from the robotModel, this just increases the searching that we have to do

Reply:
The thread should be created from the controller not the model since v_main is a controller function. It was also in the original comments in ControllerClass.py

Reply:
That's not what I was asking. I'm wondering why the thread is disassociated from the robotModel. Meaning the robotModel no longer has a thread

Reply:
Ah I see what you mean. That was just the way I implemented it. I can change it to re-attach the thread to the robotModel after creating it in the controller.

Reply:
Yeah, and then we won't have to do the extra searching, too

Reply:
@Mitchell-Conrad-20
Updated in latest commit, check it out and resolve if approved.